daccs_node_registry/update.py

```diff
@@ -49,13 +50,28 @@ def update_registry() -> None:
     for name, data in registry.items():
         org_data = deepcopy(data)
         # set services and version
+        services_url = None
+        version_url = None
+        for link in data["links"]:
+            if link["rel"] == "collection":
+                services_url = link["href"]
+            elif link["rel"] == "version":
+                version_url = link["href"]
+        if services_url is None:
+            registry[name]["status"] = "invalid_configuration"
+            sys.stderr.write(f"invalid configuration for Node named {name}: missing 'collection' link")
+            continue
+        if version_url is None:
+            registry[name]["status"] = "invalid_configuration"
+            sys.stderr.write(f"invalid configuration for Node named {name}: missing 'version' link")
+            continue
```

Comment:
Instead, the `jsonschema.validate(instance=registry, schema=schema)` could be performed at the start to avoid extra/custom checks. After validation, the keys can be accessed directly.

Reply:
Good point, we can even just skip that entirely since the schema is assumed to be initially valid (and if it isn't then I'm happy for the script to fail catastrophically)

Reply:
The advantage of doing `jsonschema.validate` is that it should print out an exception with the specific reason it failed.
This can be convenient when the error is caused by an invalid value, like a failing regex pattern, and not a missing value.

Reply:
We're doing that validation before the registry is updated so that the errors are reported when they are first introduced. (see the last try/except/else in this function)

The only other place that this could be updated into an invalid state would be from a pull request so I'll throw in an additional test for that case as well. Then all of our bases will be covered.